examples/models/export_hf_model.py

```diff
@@ -0,0 +1,178 @@
+    # architecture: gemma
+    ("google/gemma-2b",),
+    ("google/gemma-7b",),
+    # unknown
+    ("microsoft/biogpt", "GOOD"),
+    ("deepseek-ai/deepseek-llm-7b-chat",),
+    ("01-ai/Yi-6B",),
+    ("BAAI/Aquila-7B", "GOOD"),
+    ("BAAI/Aquila2-7B", "GOOD"),
+    ("THUDM/chatglm3-6b",),
+]
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-hfm",
+        "--hf_model_repo",
+        required=False,
+        default=None,
+        help="a valid huggingface model repo name",
+    )
+    parser.add_argument("-o", "--output-dir", default=".", help="output directory")
+    parser.add_argument(
+        "-etx",
+        "--to_et_w_xnn",
+        required=False,
+        action="store_true",
+        help="export ExecuTorch with XNNPACK",
+    )
+    parser.add_argument(
+        "-sg",
+        "--skip_good",
+        required=False,
+        action="store_true",
+        help="skip good models",
+    )
+    args = parser.parse_args()
+
+    if not os.path.exists(args.output_dir):
+        print(f"Error: Output directory '{args.output_dir}' does not exist.")
+        return sys.exit(1)
+
+    hf_model_repos = (
+        HF_LLM_MODLE_REPOS if not args.hf_model_repo else [(args.hf_model_repo,)]
+    )
+
+    device = "cpu"
+    prompt = "What is your favourite condiment?"
+    result_summary = {}
+
+    for repo_config in hf_model_repos:
+        hf_model_repo, status = repo_config[:2] + ("",) * (2 - len(repo_config))
+        if args.skip_good and status == "GOOD":
+            print(f"Skip model '{hf_model_repo}'.")
+            result_summary[hf_model_repo] = "SKIPPED"
+            continue
+
+        model_name = hf_model_repo.split("/")[-1].lower()
+        try:
+            print("\n\n#######################################################")
+            print(f"Load model '{hf_model_repo}' from Hugginface.")
+            print("#######################################################")
+            model = AutoModelForCausalLM.from_pretrained(
+                hf_model_repo, trust_remote_code=True
+            ).to(device)
+            model.eval()
+            tokenizer = AutoTokenizer.from_pretrained(
+                hf_model_repo, trust_remote_code=True
+            )
+            model_inputs = (
+                tokenizer([prompt], return_tensors="pt").to(device)["input_ids"],
+            )
+
+            edge_m = export_to_edge(
+                model,
+                model_inputs,
+                edge_compile_config=EdgeCompileConfig(
+                    _check_ir_validity=False,
+                ),
+            )
+            print(f"Exported '{hf_model_repo}' successfully.")
+
+            if args.to_et_w_xnn:
+                partitioners = {}
+                partitioners[
+                    XnnpackDynamicallyQuantizedPartitioner.__name__
+                ] = XnnpackDynamicallyQuantizedPartitioner()
+                prog = edge_m.to_backend(partitioners).to_executorch()
```

Comment:
If you're having problems emitting large models, it should help to pass `ExecutorchBackendConfig(extract_constant_segment=True, extract_delegate_segments=True)` to `to_executorch()`. This will reduce the size of the JSON that is provided to `flatc`.

Reply:
@dbort W/ the suggested config emitting the gpt-neox-20b still crashes my devserver silently and I suspect if it's challenging the limit of flatc and would like to get more debug info. Would `ET_EXIR_SAVE_FLATC_INPUTS_ON_FAILURE =1` be helpful? And what is the right way to set this env var, something like `set ET_EXIR_SAVE_FLATC_INPUTS_ON_FAILURE=1 && python <my_script>`?

Reply:
> Would `ET_EXIR_SAVE_FLATC_INPUTS_ON_FAILURE =1` be helpful?

It could help debug the issue. It's an environment variable, so you can set it with `export ET_EXIR_SAVE_FLATC_INPUTS_ON_FAILURE=1` in your terminal, then running your export scripts.